### PR TITLE
Verbose verdict for checkaccess

### PR DESCRIPTION
**Pull request by Anumita**

### CODE REVIEW

File: authz/providers/azure/rbac/checkaccessreqhelper.go [hunk 1]
```diff
@@ -324,7 +326,11 @@ func ConvertCheckAccessResponse(body []byte) (*authzv1beta1.SubjectAccessReviewS
 
 	if strings.ToLower(response[0].Decision) == Allowed {
 		allowed = true
-		verdict = AccessAllowedVerdict
+		if (AzureRoleAssignment{} != response[0].AzureRoleAssignment) && (response[0].AzureRoleAssignment.Id != "" || response[0].AzureRoleAssignment.RoleDefinitionId != "" || response[0].AzureRoleAssignment.PrincipalId != "") {
```
Comment (krdhruva) [inline authz/providers/azure/rbac/checkaccessreqhelper.go:329]: should this be empty check or nil check?

Comment (Anumita) [reply authz/providers/azure/rbac/checkaccessreqhelper.go:329]: empty as while unmarshalling , those fields will never be null , they will be empty as they are of string type

Comment (krdhruva) [reply authz/providers/azure/rbac/checkaccessreqhelper.go:329]: We are fine with string having empty. What we want to make sure is we don't crash. Althogh Id is string type they are part of AzureRoleAssignment which is struct that can be nil. right?

Comment (Anumita) [reply authz/providers/azure/rbac/checkaccessreqhelper.go:329]: AzureRoleAssignment will not be nil, as json unmarshal will make it as an empty struct

Comment (krdhruva) [reply authz/providers/azure/rbac/checkaccessreqhelper.go:329]: got it. Then let's have only one message. In worst case we would print those fields as empty. Anyways it's not expected to be empty on most of the cases.

File: authz/providers/azure/rbac/checkaccessreqhelper.go [hunk 2]
```diff
@@ -27,7 +28,8 @@ import (
 )
 
 const (
-	AccessAllowedVerdict        = "Access allowed"
+	AccessAllowedVerdict        = "Access allowed by Azure RBAC"
+	AccessAllowedVerboseVerdict = "Access allowed by Azure RBAC Role Assignment %s using Role %s to user %s"
```
Comment (krdhruva) [inline authz/providers/azure/rbac/checkaccessreqhelper.go:32]: of?

**Comment by krdhruva** [comment]
we also discussed to print verdict message in log.

**Review by krdhruva — CHANGES_REQUESTED** [review]
🕐

**Review by krdhruva — APPROVED** [review]

**Pull request merged**